src/socialMediaActivity.js

```diff
@@ -1094,26 +1431,31 @@ function verifyRedditCommentVote({requestDetails = null}) {
 
 /**
  * Parse a Reddit comment vote request into an object.
- * @param requestDetails - the raw request
- * @returns - the parsed object
+ * @param {Object} requestDetails - the raw request
+ * @returns {redditActivityDetails} - the parsed object
  * @private
  */
 function extractRedditCommentVote({requestDetails = null, eventTime = null}) {
-    return new Promise((resolve, reject) => {
-        const details = {};
-        details.eventTime = eventTime;
-        details.eventType = "commentVote";
-        details.vote = requestDetails.requestBody.formData.dir[0];
-        details.commentId = requestDetails.requestBody.formData.id[0];
-
-        getRedditThingContents(details.commentId).then((hydratedComment) => {
-            details.postId = hydratedComment.data.children[0].data.link_id;
-            details.commentContents = hydratedComment;
-            resolve(details);
-        });
-    });
+    const details = createBaseRedditObject();
+    details.eventTime = eventTime;
+    details.eventType = "commentVote";
+    try {
+        details.voteDirection = requestDetails.requestBody.formData.dir[0];
+    } catch { //ignore
+    }
+    try {
+        details.votedCommentId = requestDetails.requestBody.formData.id[0];
+    } catch { //ignore
+    }
+
+    return details;
 }
 
+/**
+ * For a given subreddit name, return whether it is private or public.
+ * @param {string} subredditName - The name of the subreddit, without leading "r/".
+ * @returns {Promise} - resolves to a string representing the subreddit's status, or "unknown".
+ */
 export function checkSubredditStatus(subredditName) {
     if (subredditName == "") return "unknown";
     return new Promise((resolve, reject) => {
```

Comment:
Here and throughout, you don't need to wrap `fetch` in a new `Promise`, because it already starts a `Promise` chain.